src/main/java/org/junitpioneer/jupiter/PioneerUtils.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+/**
+ * Pioneer-internal utility class.
+ */
+class PioneerUtils {
+
+	private PioneerUtils() {
+		// private constructor to prevent instantiation of utility class
+	}
+
+	/**
+	 * Determines whether an annotation of any of the specified {@code annotationTypes}
+	 * is either <em>present</em>, <em>indirectly present</em>, or <em>meta-present</em>
+	 * on the test element (method or class) or any enclosing class belonging to the
+	 * specified {@code context}.
+	 */
+	public static boolean annotationsPresent(ExtensionContext context, Class<? extends Annotation>... annotationTypes) {
```

Comment:
i am not sure about the naming here,` isAnyAnnotationPresent` would be more representing, right now i would expect that all annotations are present